internal/appident/get_appidentifier.go

```diff
@@ -31,23 +36,23 @@ func GetAppIdentifier(appDir string, m *manifest.Manifest, orgSlug string, appSl
 
 	ai := AppIdentifier{OrganizationSlug: orgSlug, AppSlug: appSlug}
 
-	if orgSlug == "" {
+	if ai.OrganizationSlug == "" {
 		return AppIdentifier{}, ErrMissingOrganizationSlug
 	}
 
-	if !validate.IsValidIdentifier(orgSlug) {
+	if !validate.IsValidIdentifier(ai.OrganizationSlug) {
 		return ai, ErrInvalidOrganizationSlug
 	}
 
-	if appSlug == "" {
+	if ai.AppSlug == "" {
 		return AppIdentifier{}, ErrMissingAppSlug
 	}
 
-	if !validate.IsValidIdentifier(appSlug) {
+	if !validate.IsValidIdentifier(ai.AppSlug) {
 		return ai, ErrInvalidAppSlug
 	}
 
-	return AppIdentifier{OrganizationSlug: orgSlug, AppSlug: appSlug}, nil
+	return ai, nil
```

Comment:
The implementation of that function was a bit inconsistent, good catch!

But I think I would prefer a change where either:

1. The `ai` variable is removed entirely, since otherwise we are kind of instantiating an invalid value
2. Or, validation logic  is factored into a "private" method `AppIdentifier.validate`

Reply:
Extracted validation logic into a  method.